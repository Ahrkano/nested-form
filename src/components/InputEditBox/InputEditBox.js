import React from 'react';
import Aux from '../../hoc/Auxiliary/Auxiliary';
import './InputEditBox.css'

import InputButton from '../Buttons/InputButton/InputButton';
    
const inputEditBox = (props) => {
    const marginLeft = 40 * props.level + 'px';

    let condition = null;
    if(props.condition !== 'noCondition') {
        if(props.parentType === 'text') {
            condition = (
                <Aux>
                    <select 
                        id="questionType"
                        value={props.condition} 
                        onChange={(event) => props.onInputChange(event, props.id, 'condition')}>
                        <option value="equals">Equals</option>
                    </select>
                    <input
                        type="text" 
                        id="questionInput" 
                        value={props.conditionValue} 
                        onChange={(event) => props.onInputChange(event, props.id, 'conditionValue')} />
                </Aux>
            );
        } else if(props.parentType === 'yesNo') {
            condition = (
                <Aux>
                    <select 
                        id="questionType"
                        value={props.condition} 
                        onChange={(event) => props.onInputChange(event, props.id, 'condition')}>
                        <option value="equals">Equals</option>
                    </select>
                    <select 
                        id="questionType"
                        value={props.conditionValue} 
                        onChange={(event) => props.onInputChange(event, props.id, 'conditionValue')}>
                        <option value="yes">Yes</option>
                        <option value="no">No</option>
                    </select>
                </Aux>
            );            
        } else {
            condition = (
                <Aux>
                    <select 
                        id="questionType"
                        value={props.condition} 
                        onChange={(event) => props.onInputChange(event, props.id, 'condition')}>
                        <option value="equals">Equals</option>
                        <option value="greater">Greater than</option>
                        <option value="less">Less than</option>
                    </select>
                    <input 
                        type="number"
                        id="questionInput" 
                        value={props.conditionValue} 
                        onChange={(event) => props.onInputChange(event, props.id, 'conditionValue')} />
                </Aux>
            );
        }
    }

    const questionInput = (
        <div className="InputEditBox__question-value">
            <label htmlFor="questionInput">Question: </label>
            <input 
                id="questionInput" 
                value={props.value} 
                onChange={(event) => props.onInputChange(event, props.id, 'question')} />
        </div>
    );

    const type = (
        <div className="InputEditBox__question-type">
            <label htmlFor="questionType">Type: </label>
            <select 
                id="questionType"
                value={props.type} 
                onChange={(event) => props.onInputChange(event, props.id, 'type')}>
                <option value="yesNo">Yes / No</option>
                <option value="text">Text</option>
                <option value="number">Number</option>
            </select>
        </div>
    );

    return (
        <div className="InputEditBox" style={{ marginLeft: marginLeft }}>
            <div className="InputEditBox__content">
                <h5 className="InputEditBox__question-id">{props.id}</h5>
                {props.condition !== 'noCondition' ?
                    <div className="InputEditBox__condition">
                        <label>Condition: </label>
                        {condition}
                    </div>
                    : null}

                {questionInput}
                {type}
            </div>
            <div className="InputEditBox__buttons">
                <InputButton
                    className="InputButton__delete"
                    onButtonClick={() => props.onInputDeletion(props.id, props.parent)}>
                    Delete
                </InputButton>
                <InputButton 
                    className="InputButton__add-sub-input"
                    onButtonClick={() => props.onSubInputAddition(props.id, props.level, props.type)}>
                    Add Sub-Input
                </InputButton>
            </div>
        </div>
    );
}

export default inputEditBox;