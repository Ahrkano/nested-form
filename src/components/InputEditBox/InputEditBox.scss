@import 'shared/_colors.scss';

.InputEditBox {
    display: flex;
    justify-content: space-between;
    width: 450px;
    height: 100px;
    margin: 1rem 0;
    padding: 1rem;
    border: 2px solid rgba($color-primary, .8);
    border-radius: 10px;

    color: hsl(0, 0%, 40%);

    & label {
        color: #fff;
        background-color: hsl(0,0%, 70%);
        padding: 2px;
        border-radius: 3px;
        width: 100px;
    }

    & input,
    & select {
        padding: 2px;
        margin-bottom: 5px;
        margin-left: 3px;
        border-radius: 3px;
    }

    &__question-id {
        margin: 0;
        margin-bottom: 10px;
        color: hsl(0, 0%, 80%);
    }

    &__content {
        width: 75%;
    }

    &__buttons {
        display: flex;
        flex-direction: column;
        width: 25%;
    }
}