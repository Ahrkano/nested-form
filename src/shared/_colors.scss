$color-primary: #0089BA;
$color-secondary: #D5CABD;
$color-warning: rgba(#d9534f, .8);
$color-warning-hover: #de716e;
$color-success: #95d095;
$color-success-hover: #5cb85c;