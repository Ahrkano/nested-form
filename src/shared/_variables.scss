$navbar-height: 50px;
$navbar-width: 500px;