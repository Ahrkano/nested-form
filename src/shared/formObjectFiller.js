export const formObjectFiller = function(questionArray) {
    if (questionArray) {
        const formObject = {}
        // populating formObject with questionObjects
        questionArray.forEach(questionItem => {
            formObject[questionItem.id] = {
                conditionType: questionItem.condition === 'noCondition' ? 'root' : 'conditional',
                parentId: questionItem.parentId,
                question: questionItem.question,
                inputType: questionItem.type,
                answer: '',
                level: questionItem.anchorLevel,
                conditionalQuestions: {}
            }; 
        });

        questionArray.forEach(questionItem => {

            if (questionItem.condition !== 'noCondition') {
                formObject[questionItem.parentId].conditionalQuestions[questionItem.id] = {
                    type: questionItem.condition,
                    value: questionItem.conditionValue
                };
            }
        });

        return formObject;
    }
}

export const rootQuestionsOrderArray = function(questionArray) {
    if (questionArray) {
        const rootQuestionsOrder = [];

        questionArray.forEach(questionItem => {
            if(questionItem.parentId === 'rootNode') {
                rootQuestionsOrder.push(questionItem.id);
            }
        });

        return rootQuestionsOrder;
    }
}