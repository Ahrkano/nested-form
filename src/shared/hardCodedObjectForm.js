export const objectForm = {
    question_C1: {
        answer: "",
        conditionType: "conditional",
        conditionalQuestions: {},
        inputType: "yesNo",
        parentId: "question_C0",
        question: "Third's question subinput"
    },
    question_A0: {
        answer: "",
        conditionType: "root",
        conditionalQuestions: {
            question_A1: {type: "equals", value: "yes"}
        },
        inputType: "yesNo",
        parentId: "rootNode",
        question: "First root question: choose yes",
    },
    question_A2: {
        answer: "",
        conditionType: "conditional",
        conditionalQuestions: {},
        inputType: "yesNo",
        parentId: "question_A1",
        question: "Third question",
    },
    question_A1: {
        answer: "",
        conditionType: "conditional",
        conditionalQuestions: {
            question_A2: {type: "equals", value: "second"}
        },
        inputType: "text",
        parentId: "question_A0",
        question: "Second question: input 'second'"
    },
    question_C0: {
        answer: "",
        conditionType: "root",
        conditionalQuestions: {
            question_C1: {type: "equals", value: "4"}
        },
        inputType: "number",
        parentId: "rootNode",
        question: "Third root question: input '4'"
    },
    question_B0: {
        answer: "",
        conditionType: "root",
        conditionalQuestions: {},
        inputType: "yesNo",
        parentId: "rootNode",
        question: "Second root question",
    }
}

export const rootQuestionsOrder = ["question_A0", "question_B0", "question_C0"];