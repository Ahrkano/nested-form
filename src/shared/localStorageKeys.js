export const MAIN_KEY = 'nestedFormData##';
