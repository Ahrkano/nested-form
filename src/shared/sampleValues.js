export const formObjectInstance = {
    question_8cbc0838: {
        conditionType: 'root',
        condition: 'noCondition',
        conditionValue: 'noCondition',
        parentId: 'rootNode',
        parentType: 'root',
        question: 'Write good',
        inputType: 'text',
        answer: '',
        level: 1,
        conditionalQuestions: {
            question_9a7475a1: {
                type: 'equals',
                value: 'good'
            }
        }
    },
    question_9a7475a1: {
        conditionType: 'conditional',
        condition: 'equals',
        conditionValue: 'good',
        parentId: 'question_8cbc0838',
        parentType: 'text',
        question: 'Click yes',
        inputType: 'yesNo',
        answer: '',
        level: 2,
        conditionalQuestions: {}
    },
    question_3ad9522f: {
        conditionType: 'root',
        condition: 'noCondition',
        conditionValue: 'noCondition',
        parentId: 'rootNode',
        parentType: 'root',
        question: 'Some question',
        inputType: 'yesNo',
        answer: '',
        level: 1,
        conditionalQuestions: {}
    }
};

export const rootQuestionOrderInstance = ['question_8cbc0838', 'question_3ad9522f'];
