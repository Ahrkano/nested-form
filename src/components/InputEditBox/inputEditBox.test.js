import React from 'react';
import renderer from 'react-test-renderer';
import InputEditBox from './InputEditBox';
import { formObjectInstance, rootQuestionOrderInstance } from '../../shared/sampleValues';

const questionId = rootQuestionOrderInstance[0];

describe('InputEditBox snapshot', () => {
    test('InputEditBox snapshot test', () => {
        const component = renderer.create(
            <InputEditBox
                key={questionId}
                id={questionId}
                value={formObjectInstance[questionId].question}
                type={formObjectInstance[questionId].inputType}
                parentType={formObjectInstance[questionId].parentType}
                condition={formObjectInstance[questionId].condition}
                conditionValue={formObjectInstance[questionId].conditionValue}
                level={formObjectInstance[questionId].level}
                parent={formObjectInstance[questionId].parentId}
                onInputChange={() => {}}
                onSubInputAddition={() => {}}
                onInputDeletion={() => {}}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
