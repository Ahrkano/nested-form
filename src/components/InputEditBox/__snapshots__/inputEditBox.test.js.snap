// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputEditBox snapshot InputEditBox snapshot test 1`] = `
<div
  className="InputEditBox"
  style={
    Object {
      "marginLeft": "40px",
    }
  }
>
  <div
    className="InputEditBox__content"
  >
    <h5
      className="InputEditBox__question-id"
    >
      question_8cbc0838
    </h5>
    <div
      className="InputEditBox__question-value"
    >
      <label
        htmlFor="questionInput"
      >
        Question: 
      </label>
      <input
        id="questionInput"
        onChange={[Function]}
        value="Write good"
      />
    </div>
    <div
      className="InputEditBox__question-type"
    >
      <label
        htmlFor="questionType"
      >
        Type: 
      </label>
      <select
        id="questionType"
        onChange={[Function]}
        value="text"
      >
        <option
          value="yesNo"
        >
          Yes / No
        </option>
        <option
          value="text"
        >
          Text
        </option>
        <option
          value="number"
        >
          Number
        </option>
      </select>
    </div>
  </div>
  <div
    className="InputEditBox__buttons"
  >
    <button
      className="InputButton__delete"
      onClick={[Function]}
    >
      Delete
    </button>
    <button
      className="InputButton__add-sub-input"
      onClick={[Function]}
    >
      Add Sub-Input
    </button>
  </div>
</div>
`;
